modules/core/src/main/java/org/locationtech/jts/geom/prep/BasicPreparedGeometry.java

```diff
@@ -31,18 +29,43 @@
  * @author Martin Davis
  *
  */
-class BasicPreparedGeometry 
+class BasicPreparedGeometry extends Geometry
   implements PreparedGeometry
 {
   private final Geometry baseGeom;
   private final List representativePts;  // List<Coordinate>
 
   public BasicPreparedGeometry(Geometry geom) 
   {
+    super(geom.getFactory());
     baseGeom = geom;
     representativePts = ComponentCoordinateExtracter.getCoordinates(geom);
   }
 
+  public Geometry copy() {
+    return baseGeom.copy();
+  }
+
+  public void normalize() {
+    baseGeom.normalize();
+  }
+
+  protected Envelope computeEnvelopeInternal() {
+    return baseGeom.getEnvelopeInternal();
```

Comment:
should this be `baseGeom.computeEnvelopeInternal` ?

Reply:
Should but. But it that method is protected

Reply:
But can't a subclass override a protected method in the parent class?